public/app/plugins/panel/stat/common.ts

```diff
@@ -107,3 +112,55 @@ export function addOrientationOption<T extends SingleStatBaseOptions>(
     defaultValue: VizOrientation.Auto,
   });
 }
+
+export function formatDisplayValuesWithCustomUnits(
+  fieldValues: FieldDisplay[],
+  config: PanelFieldConfig
+): FieldDisplay[] {
+  const { prefix, suffix } = config;
+
+  if (!prefix && !suffix) {
+    return fieldValues;
+  }
+
+  return fieldValues.map((fieldValue) => {
+    const fieldType = fieldValue?.sourceField?.type;
+    // Test for FieldType.number, since that is the only type on which formatting is enforced
+    if (fieldType === FieldType.number) {
+      const { display } = fieldValue;
+      return {
+        ...fieldValue,
+        display: {
+          ...display,
+          // Test also for nullishness, otherwise "undefined" will be concatenated into the string
+          prefix: `${prefix ?? ''}${display.prefix ?? ''}`,
+          suffix: `${display.suffix ?? ''}${suffix ?? ''}`,
+        },
+      };
+    }
```

Comment:
This might be a little cleaner if we split this out into two separate consts that got assigned as part of the returned object i.e.

```suggestion
    if (fieldType === FieldType.number) {
      const { display } = fieldValue;
      
      // Test also for nullishness, otherwise "undefined" will be concatenated into the string
      const customPrefix = `${prefix ?? ''}${display.prefix ?? ''}`
      const customSuffix = `${display.suffix ?? ''}${suffix ?? ''}`
      
      return {
        ...fieldValue,
        display: {
          ...display,
          prefix: customPrefix,
          suffix: customSuffix,
        },
      };
    }
```